forgerock-auth/src/main/java/org/forgerock/android/auth/FROptions.kt

```diff
@@ -0,0 +1,234 @@
+                    && old?.server == new?.server
+                    && old?.sslPinning == new?.sslPinning
+                    && old?.service == new?.service
+                    && old?.urlPath == new?.urlPath
+                    && old?.logger == new?.logger
+        }
+    }
+}
+
+/**
+ * Option builder to build the SDK configuration information
+ */
+class FROptionsBuilder {
+
+    private var server: Server = Server()
+    private var oauth: OAuth = OAuth()
+    private var service: Service = Service()
+    private var urlPath: UrlPath = UrlPath()
+    private var sslPinning: SSLPinning = SSLPinning()
+    private var logger: Log = Log()
+
+    companion object {
+        @JvmStatic
+        fun build(block: FROptionsBuilder.() -> Unit): FROptions = FROptionsBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the server configurations
+     *
+     * @param block  takes the  closure to set the values for server configuration.
+     */
+    fun server(block: ServerBuilder.() -> Unit) {
+        server = ServerBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the oauth configurations
+     *
+     * @param block takes the  closure to set the values for oauth configuration.
+     */
+    fun oauth(block: OAuthBuilder.() -> Unit) {
+        oauth = OAuthBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the service configurations
+     *
+     * @param block takes the closure to set the values for service configuration.
+     */
+    fun service(block: ServiceBuilder.() -> Unit) {
+        service = ServiceBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the endpoints
+     *
+     * @param block takes the  closure to set the values for urls.
+     */
+    fun urlPath(block: UrlPathBuilder.() -> Unit) {
+        urlPath = UrlPathBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the ssl pinning
+     *
+     * @param block takes the builder closure to set the values for ssl pinning.
+     */
+    fun sslPinning(block: SSLPinningBuilder.() -> Unit) {
+        sslPinning = SSLPinningBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the custom logger
+     *
+     * @param block takes the builder closure to configure custom logger.
+     */
+    fun logger(block: LoggerBuilder.() -> Unit) {
+        logger = LoggerBuilder().apply(block).build()
+    }
+
+    private fun build(): FROptions {
+        return FROptions(server, oauth, service, urlPath, sslPinning, logger)
+    }
+
+}
+
+/**
+ * Data class for the server configurations
+ */
+data class Server(val url: String? = "",
+                  val realm: String? = "root",
+                  val timeout: Int = 30,
+                  val cookieName: String? = "iPlanetDirectoryPro")
+
+/**
+ * Server builder to build the SDK configuration information specific to server
+ */
+class ServerBuilder {
+    var url: String? = ""
+    var realm: String? = "root"
+    var timeout: Int = 30
```

Comment:
Default timeout in iOS is 60s

Reply:
i would like to keep the same as 30, since its not been changed , i know its not aligned . not sure it will create side effect for the users and for the backward compatibility let's not change this

Reply:
Andy and i talked about this, we are ok with this to be not aligned